src/transform-sdk/js/js_vm.cc

```diff
@@ -250,6 +251,30 @@ value::set_property(std::string_view key, const value& val) {
     return {};
 }
 
+std::expected<std::monostate, exception> value::push_back(const value& val) {
+    const size_t len = array_length();
+    constexpr auto flags = JS_PROP_C_W_E; // NOLINT
+    const int result = JS_DefinePropertyValueUint32(
+      _ctx, _underlying, len, val.raw_dup(), flags);
+    if (result < 0) {
+        return std::unexpected(exception::current(_ctx));
+    }
+    return {};
+}
```

Comment:
suggestion (non-blocking): worth putting a type check in there somewhere?

Reply:
Yeah done for most of these functions